.github/actions/run-local-api/action.yml

```diff
@@ -24,33 +20,28 @@ runs:
     using: composite
 
     steps:
-        -   name: Set up Python ${{ inputs.python_version }}
-            uses: actions/setup-python@v4
-            with:
-                python-version: ${{ inputs.python_version }}
-                cache: "pip"
-
-        -   name: Copy pip requirements
-            run: cp api/requirements.txt .
-            shell: bash
-
-        -   name: Install Dependencies
-            run: |
-                python -m pip install --upgrade pip
-                pip install -r requirements.txt
-            shell: bash
+        - name: Build temporary Docker image for running the API
+          working-directory: api
+          run: |
+              docker build -t flagsmith/flagsmith-api:e2e .
```

Comment:
Hmm... I have no idea how github workflows are run - this doesn't seem immune to collisions? Should we use some kind of hash? Or the commit sha? 

Also, this can all go on one line I think? 

```
run: docker build -t flagsmith/flagsmith-api:e2e .
```

Reply:
Ths builds a new, local docker image each time.